### PR TITLE
[`flake8-async`] Update `ASYNC116` to match upstream

### PR DESCRIPTION
**Pull request by augustelalande**
## Summary

Update the name of `ASYNC116` to match [upstream](https://flake8-async.readthedocs.io/en/latest/rules.html).

Also update the functionality to match upstream by adding support for `anyio` (gated behind preview). `asyncio` does not have a `sleep_forever` equivalent.

## Test Plan

Added tests for `anyio`

### CODE REVIEW

**Review by charliermarsh — APPROVED** [review]
Excellent, thanks.

File: crates/ruff_linter/src/rules/flake8_async/rules/long_sleep_not_forever.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        {
+            #[allow(clippy::cast_precision_loss)]
+            if *float_value <= one_day_in_secs as f64 {
+                return;
+            }
+        }
+        Number::Complex { .. } => return,
+    }
+
+    let module = AsyncModule::try_from(&qualified_name).unwrap();
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/flake8_async/rules/long_sleep_not_forever.rs:111]: Maybe we extract this simultaneously with the `!matches!(qualified_name.segments(), ["trio" | "anyio", "sleep"])` above, to make the `unwrap` unnecessary?

File: crates/ruff_linter/src/rules/flake8_async/rules/long_sleep_not_forever.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    if matches!(checker.settings.preview, PreviewMode::Disabled) {
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/flake8_async/rules/long_sleep_not_forever.rs:112]: I think we have an `.is_disabled()` method.

Comment (augustelalande) [reply crates/ruff_linter/src/rules/flake8_async/rules/long_sleep_not_forever.rs]: Done

File: crates/ruff_linter/src/rules/flake8_async/rules/long_sleep_not_forever.rs (hunk 1 above)
Comment (augustelalande) [reply crates/ruff_linter/src/rules/flake8_async/rules/long_sleep_not_forever.rs:110]: I'm not sure what pattern you're suggesting

Comment (zanieb) [reply crates/ruff_linter/src/rules/flake8_async/rules/long_sleep_not_forever.rs:110]: Maybe something like

```rust
let module = match qualfied_name.segments() {
   ["trio", "sleep"] => AsyncModule::Trio,
   ["anyio", "sleep"] => AsyncModle::Anyio,
   _ => return;
};
```

or

```rust
let Some(module) = AsyncModule::try_from(&qualified_name) else { return; };
if !matches!(qualified_name.segments(), [_, "sleep"]) { return; };
```

**Pull request merged**